src/components/BaseInput/PBaseInput.vue

```diff
@@ -74,11 +81,22 @@
   focus-within:ring-offset-focus-ring-offset
   rounded-default
   font-normal;
-  background-color: var(--p-color-input-bg);
   border-color: var(--p-color-input-border);
+  background-color: var(--p-color-input-bg);
   color: var(--p-color-input-text);
 }
 
+.p-base-input--small { @apply
+  text-xs
+  p-0
+}
+
+.p-base-input--small input { @apply
```

Comment:
My guess that this isn't enough. Base input is used with many different elements for the "input" such as `button` and `select`. 

Also increasing specificity here is going to cause problems because then a component implementing base-input will need a higher specificity in order to say changes its padding. 

I think a possible solution here would be to make a `p-base-input__control` block class and then add the modifier `p-base-input__control--small`. Then both classes can be added to what is currently called `attrsWithDisabled` which could probably get renamed to `attrsForControl` or something.